torchao/prototype/intx/Intx.py

```diff
@@ -0,0 +1,255 @@
+import functools
+import math
+from collections import defaultdict
+from typing import Any, Callable, Dict, Optional, Tuple, Union, List
+
+import torch
+from torch._dynamo.comptime import comptime
+
+from torch.utils._python_dispatch import return_and_correct_aliasing
+from .bitpacking import pack, unpack, numbits
+from torchao.dtypes.utils import (
+    _implements,
+    _ATEN_OP_OR_TORCH_FN_TABLE,
+    _register_layout_cls,
+    _get_layout_tensor_constructor,
+)
+
+
+aten = torch.ops.aten
+def implements(aten_ops_or_torch_fn):
+    return _implements(IntxTensor, aten_ops_or_torch_fn)
+
+def get_layout_tensor_constructor(layout_type_class: type(LayoutType)):
+    return _get_layout_tensor_constructor(AffineQuantizedTensor, layout_type_class)
+
+@dataclass(frozen=True)
+class IntxLayoutType(LayoutType):
```

Comment:
OK it's fine to have this as `IntxLayoutType` for now but I feel we probably want to make this default after it's more mature

Reply:
What do you mean make it default?